### DISCUSSION TITLE
A guide to implement harmoniser

### DISCUSSION THREAD

**Discussion by EasperOne** [EBISPOT/gwas-sumstats-harmoniser#102] [General]
Hi there,

I'm new here and really want to test the harmonisation pipeline. But I'm stuck. Hope that somebody could walk me through. According to the guide in README file. To use this pipeline: 

**1. We first need to run this to download and test.**

```
nextflow run  EBISPOT/gwas-sumstats-harmoniser -profile test,conda
```

**2. Then about the problematic reference preparation:**
 From what I understand, the ref could be prepared from my own by run this (denoted as CODE 2) and specify my reference directory right?
```
nextflow run  EBISPOT/gwas-sumstats-harmoniser \
--reference \
--ref 'full path my reference' \
-profile cluster,conda
```
OR, I could download the prepared ref from your FTP server which is shown here
![image](https://github.com/user-attachments/assets/1aacfa3a-d34f-4249-9612-a0f675280025)

OR, I could prepare the ref from your default `--remote_vcf_location` and `--remote_ensembl_variation` by running this
```
nextflow run EBISPOT/gwas-sumstats-harmoniser \
--reference \
-profile cluster,conda
```
So, if I download the ref which I did, I won't have to do the ref preparation right? Because whenever I tried to prepare ref by running the CODE 2 on the downloaded ref, I always failed the stage get_variation_table and got stuck at make_local_synonyms_table.
I took a glance at your code and saw that the make_local_synonyms_table stage output a rsID.sql file, which is already there in the downloaded ref.

**3. Run harmonisation:**
I ran
```
nextflow run EBISPOT/gwas-sumstats-harmoniser \
--harm \
--ref 'full path to the downloaded ref' \
--file 'full path to GCST1.tsv' \ (which is also your test case)
-profile cluster,conda
```
But failed at map_to_build and it kept trying that stage again and again.

I don't know where I was wrong. It would be so so so kind of you to help me with this situation.

Best,
Khoa

**Comment by jiyue1214** [comment]
Firstly, thank you for trying our pipeline!

We are close to releasing an update that addresses these issues, along with our formal documentation, which we hope will be helpful to you. I’ll let you know as soon as the new release is ready.

Additionally, the resource bundle contains all the reference files you’ll need, allowing you to skip CODE 2. We also suggest using Docker or Singularity to ensure a compatible environment to run the pipeline. If you’re still experiencing issues, could you please provide the .nextflow.log file? This will help us better understand what happened and provide further assistance.
